nservicebus/scheduling/index.md

```diff
@@ -13,9 +13,11 @@ The Scheduler is a lightweight/non-durable API that helps schedule a task that n
 
 {{WARNING:
 
-Scheduling a task depends on [delayed delivery](/nservicebus/messaging/delayed-delivery.md). If the delayed delivery operation fails, the Scheduler will be interrupted and a `ScheduledTask` message will be forwarded to the error queue. When this happens the scheduled task will stop executing unless the `ScheduledTask` message is retried.
+Under certain failure scenarios, the scheduled task can stop firing until the endpoint is restarted. 
```

Comment:
This needs explanation, what is 'certain failure scenarios'?

Reply:
You are hinting that it operates as at-most-once triggering, if that is the case we should remove the feature ASAP.

Reply:
@ramonsmits full context is in https://github.com/Particular/NServiceBus/issues/5591.

Long story short when using Outbox and not using the same db with SQL Transport it's quite possible for tasks to stop executing. The user is hitting that scenario ~1 a week.

Reply:
@tmasternak So the problem is that the schedule message gets moved to the error queue? It isn't actually lost.

Maybe adding a [custom recovery policy](https://docs.particular.net/nservicebus/recoverability/custom-recoverability-policy#implement-a-custom-policy-full-customization) for schedule message helps in to not move it to the error queue?

Reply:
> So the problem is that the schedule message gets moved to the error queue

@ramonsmits correct, the message gets moved to the error queue and stays there until handled manually. I think that a custom recovery policy could work here.

Reply:
Any failures in the scheduler lambda will NOT result in any retry logic thus considered a loss:

- https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Scheduling/DefaultScheduler.cs#L41

The problem is that supplied context is wrong. That should be a context based on `IMessageSession`. With this session, a customer can do its own retry mechanism if required.

- https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Scheduling/DefaultScheduler.cs#L36

The "next" occurence is controlled with a send that happens before the execution of the task. Any failure here would result in transport recoverability

- https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Scheduling/DefaultScheduler.cs#L56

The BIGGEST problem is scale-out, brokers aren't supported.

IMHO, we should mark as Obsolete in our code base ASAP.

Reply:
@ramonsmits I agree that we should obsolete the scheduler.